interview-questions.md

```diff
@@ -8,48 +8,65 @@ Then, research the question to expand on your answer. Even if you feel you have
+Your answer: JSX is a feature provided by React that allows us to write JavaScript within HTML. It allows us to combine our JS logic and HTML UI in one file. Without it, we would have to do all our logic in JavaScript and our formatting in HTML, and then link the two together.
 
-Researched answer:
+JavaScript can be written in JSX by using curly brackets '{}', while HTML uses tags as it normally does. Here is an example of JavaScript logic inside HTML:
+
+<h1>{javaScriptFunction}</h1>
+
+Researched answer: JSX is a syntax extension for JavaScript that allows users to write HTML in JavaScript files. It was created and used by Facebook to build more dynamic and iteractive UI's with React's library. By combining both languages, this allows coders to write dynamic code without having to switch back and forth between HTML and JavaScript file.
 
 1. What is yarn? What file(s) are modified when you run the command yarn in your terminal?
 
-Your answer:
+Your answer: Yarn is a package manager that can be executed via terminal. It has different commands that execute different actions. By using some of its features, it can add files to your text editor to perform built in tasks. So far I've used it features to run jest test, creat React apps, and to view my React app on a local browser.
+
+Here are some examples of yarn commands:
+$ yarn jest
+$ yarn create react-app
+$ yarn start
 
 Researched answer:
 
+Yarn is a package manager and a project manager. It replaces the default JavaScript package manager node package manager (npm). Like React, it was also created by Facebook. By executing yarn in your terminal, you can install dependencies into your projects. Yarn can be used to install React and it's library into your project; allowing you to use its features such as JSX.
+
 3. What is a React component?
 
-Your answer:
+Your answer: I view React components as templates for information. Creating containers of data can become extremley repetitive and redundant, and it also requires a lot of line of code which can make your code look messy. React components allow you to make "blank" templates and call them over and over again on your page. The data and logic can be easily interchangable, making them highly dynamic. 
 
-Researched answer:
+One example I like to think of when referring to React components is a media player on a streaming service such as Spotify. The media player has a album picture, play/pause button, and a back/next button. By creating just one component for this template, you can then pass the information of millions of songs into it.
+
+Researched answer: A React component is a reusable building block for creating UI. A component is a self-contained block of code, that can be plugged in and used with other components. Components can recieve data through props, which allow them to operate like functions with arguments. As data changes, components will also dynamically display those changes while still maintaining their structure.
```

Comment:
Love the examples in this answer and definitely can be thought of as a building block. 💯